scripts/python-skywater-pdk/skywater_pdk/netlistsvg-generate.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 The SkyWater PDK Authors.
+#
+# Use of this source code is governed by the Apache 2.0
+# license that can be found in the LICENSE file or at
+# https://www.apache.org/licenses/LICENSE-2.0
+#
+# SPDX-License-Identifier: Apache-2.0
+
+
+import csv
+import json
+import os
+import sys
+
+
+# prerequisities:   netlistsvg
+# input:            paths to cell dirs, containing Yosys netlist 
+# output:           generates [cell_prefix].schematic.svg
+# example usage 1:  ./netlistsvg-generate.py \
+#                   ../../../libraries/sky130_fd_sc_ms/latest/cells/a2111o \
+#                   ../../../libraries/sky130_fd_sc_ms/latest/cells/a2111oi
+# example usage 2:  ./netlistsvg-generate.py ALLLIBS
+
+def outfile(cellpath, define_data, ftype='', extra='', exists=False):
+    fname = define_data['name'].lower().replace('$', '_')
+    if ftype:
+        ftype = '.'+ftype
+    outpath = os.path.join(cellpath, f'{define_data["file_prefix"]}{extra}{ftype}.svg')
+    if exists is None:
+        pass
+    elif not exists:
+        #assert not os.path.exists(outpath), "Refusing to overwrite existing file:"+outpath
+        print("Creating", outpath)
+    elif exists:
+        assert os.path.exists(outpath), "Missing required:"+outpath
+    return outpath
+
+
+def write_netlistsvg(cellpath, define_data):
+    netlist_json = os.path.join(cellpath, define_data['file_prefix']+'.json')
+    if not os.path.exists(netlist_json):
+        print("No " + define_data['file_prefix'] + ".json in", cellpath)
+    assert os.path.exists(netlist_json), netlist_json
+    outpath = outfile(cellpath, define_data, 'schematic')
+    oscmd = 'netlistsvg ' + netlist_json + ' -o ' + outpath
+    r = os.system(oscmd)>>8
```

Comment:
WTF os `>>8` here?

Please use `subprocess` module here.

Reply:
Filters out return code: `os.system` returns `[exit_code][terminating_signal_id]`.
Irrelevant now, as moved to `subprocess.call`
Tried `subprocess.Popen` - the script passes way faster, but leaves lots of processing in the background and doesn't keep track of potential `netlistsvg` crashes.